src/lib/journey/callbacks/recaptcha/recaptcha.svelte

```diff
@@ -0,0 +1,79 @@
+<script lang="ts">
+  import type { ReCaptchaCallback } from '@forgerock/javascript-sdk';
+  import type { z } from 'zod';
+  import { onMount } from 'svelte';
+
+  import type { SelfSubmitFunction, StepMetadata } from '$journey/journey.interfaces';
+  import type { styleSchema } from '$lib/style.store';
+  import type { Maybe } from '$lib/interfaces';
+
+  export let callback: Maybe<ReCaptchaCallback>;
+  export const selfSubmitFunction: Maybe<SelfSubmitFunction> = null;
+  export const stepMetadata: Maybe<StepMetadata> = null;
+  export const style: z.infer<typeof styleSchema> = {};
+
+  const siteKey = callback?.getSiteKey() ?? '';
+  let recaptchaApi = `${callback?.getOutputByName<string>('captchaApiUri', '') ?? ''}`;
+  /**
+   * AM defaults the class name to g-captcha which is wrong
+   * I dont think we should be manipulating the class here,
+   * but the classname should be g-recaptcha for google
+   */
+  const recaptchaClass: string =
+    callback?.getOutputByName<string>('captchaDivClass', 'h-captcha') ?? 'h-captcha';
+
+  onMount(() => {
+    window.frHandleCaptchaError = function handleCaptchaError() {
+      callback?.setResult('');
+      if (recaptchaClass.startsWith('h')) {
+        window.hcaptcha.render(recaptchaClass, {
```

Comment:
So they are attached by the script tag below, that is correct.

The `onMount` functions only have to be "complete" in terms of definition, before we attempt to solve the recaptcha. They are "hooks" into the recaptcha essentially. If they aren't defined, it would error when it attempts to call the function.

Reply:
Okay, so you're saying the `onMount` needs to fire before the user interacts with the Captcha?

Reply:
Correct, if they interact with the captcha before this none of the callbacks (function callbacks) will fire to set the callback values